### PR TITLE
Separating out EventTracingTestConstants

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
The important change here is that the game code will not longer have a dependency on FunctionalTest.h. The game code now uses a simple blueprint library that stores constants to be used for game code tests and functional test code alike.

### CODE REVIEW

**Review by mironec — APPROVED** [review]
Seems good to me. Did you check whether this resolves the build issues? I am slightly worried that if the constants are still in SpatialGDKFunctionalTests, we will still have to include the module, which maybe will still fail, but not sure.

**Comment by MatthewSandfordImprobable** [comment]
> Seems good to me. Did you check whether this resolves the build issues? I am slightly worried that if the constants are still in SpatialGDKFunctionalTests, we will still have to include the module, which maybe will still fail, but not sure.

Giving it a run on the NFRs.

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	static FName GetReceiveOpEventName() { return ReceiveOpEventName; }
+	static FName GetSendPropertyUpdatesEventName() { return SendPropertyUpdatesEventName; }
+	static FName GetReceivePropertyUpdateEventName() { return ReceivePropertyUpdateEventName; }
+	static FName GetSendRPCEventName() { return SendRPCEventName; }
+	static FName GetProcessRPCEventName() { return ProcessRPCEventName; }
+	static FName GetComponentUpdateEventName() { return ComponentUpdateEventName; }
+	static FName GetMergeComponentUpdateEventName() { return MergeComponentUpdateEventName; }
+
+	UFUNCTION(BlueprintCallable, Category = "EventTracingTest")
+		static FName GetUserProcessRPCEventName() { return UserProcessRPCEventName; }
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h:26]:
```suggestion
	static FName GetUserProcessRPCEventName() { return UserProcessRPCEventName; }
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+	UFUNCTION(BlueprintCallable, Category = "EventTracingTest")
+		static FName GetUserReceivePropertyEventName() { return UserReceivePropertyEventName; }
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h:29]:
```suggestion
	static FName GetUserReceivePropertyEventName() { return UserReceivePropertyEventName; }
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+	UFUNCTION(BlueprintCallable, Category = "EventTracingTest")
+		static FName GetUserReceiveComponentPropertyEventName() { return UserReceiveComponentPropertyEventName; }
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h:32]:
```suggestion
	static FName GetUserReceiveComponentPropertyEventName() { return UserReceiveComponentPropertyEventName; }
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+	UFUNCTION(BlueprintCallable, Category = "EventTracingTest")
+		static FName GetUserSendPropertyEventName() { return UserSendPropertyEventName; }
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h:35]:
```suggestion
	static FName GetUserSendPropertyEventName() { return UserSendPropertyEventName; }
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+	UFUNCTION(BlueprintCallable, Category = "EventTracingTest")
+		static FName GetUserSendComponentPropertyEventName() { return UserSendComponentPropertyEventName; }
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h:38]:
```suggestion
	static FName GetUserSendComponentPropertyEventName() { return UserSendComponentPropertyEventName; }
```

File: SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+	UFUNCTION(BlueprintCallable, Category = "EventTracingTest")
+		static FName GetUserSendRPCEventName() { return UserSendRPCEventName; }
```
Comment (improbable-danny) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTestConstants.h:41]:
```suggestion
	static FName GetUserSendRPCEventName() { return UserSendRPCEventName; }
```

**Review by improbable-danny — APPROVED** [review]
👍

**Pull request merged**